components/collector/src/metric_collectors/test_cases.py

```diff
@@ -59,8 +59,9 @@ class TestCases(MetricCollector):
         "timeout": "errored",
         "warning": "errored",
     }
-    # Regular expression to identify test case ids in entity attributes:
-    TEST_CASE_KEY_RE = re.compile(r"\w+\-\d+")
+    # Regular expression to identify test case ids in entity attributes. Matches identifiers of the form BAR-123,
+    # while ensuring they are not part of longer identifiers such as FOO-BAR-123 or BAR-123-456:
+    TEST_CASE_KEY_RE = re.compile(r"(?<![-\w])\b[A-Za-z]+\-\d+\b(?![-\w])")
```

Comment:
Shouldn't this be a metric source config parameter?

Reply:
Hmm, maybe. But for now we only support Jira ids, so we don't need different regular expressions per test case source yet.